process/transaction/shardProcess.go

```diff
@@ -613,27 +621,35 @@ func (txProc *txProcessor) finishExecutionOfRelayedTx(
 		return vmcommon.Ok, nil
 	}
 
-	err = txProc.addFeeAndValueToDest(acntDst, tx.Value, computedFees.remainingFee)
+	err = txProc.addFeeAndValueToDest(acntDst, relayedValue, computedFees.remainingFee)
 	if err != nil {
 		return 0, err
 	}
 
-	return txProc.processUserTx(tx, userTx, tx.Value, tx.Nonce, txHash)
+	relayedNonce := tx.Nonce
+	relayerAddr := tx.SndAddr
+	if isUserTxOfRelayedV3 && !check.IfNil(relayerAcnt) {
+		relayedNonce = relayerAcnt.GetNonce() - 1 // nonce already increased inside processTxAtRelayer
```

Comment:
yeah. but it is increased for the relayer. which is not really good.

In case of relayedV3 increase nonce should happen only at the user - and it has to be exactly as its transaction nonce has.